packages/in-app-notification-manager/src/app.ts

```diff
@@ -0,0 +1,45 @@
+import {
+  applicationAuditBeginMiddleware,
+  applicationAuditEndMiddleware,
+} from "pagopa-interop-application-audit";
+import {
+  authenticationMiddleware,
+  contextMiddleware,
+  loggerMiddleware,
+  zodiosCtx,
+} from "pagopa-interop-commons";
+import { serviceName as modelsServiceName } from "pagopa-interop-models";
+import { isFeatureFlagEnabled } from "pagopa-interop-commons";
+import { config } from "./config/config.js";
+import healthRouter from "./routers/healthRouter.js";
+import { notificationRouter } from "./routers/notificationRouter.js";
+import { InAppNotificationService } from "./services/inAppNotificationService.js";
+
+// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+export async function createApp(service: InAppNotificationService) {
+  const serviceName = modelsServiceName.IN_APP_NOTIFICATION_MANAGER;
+  const router = notificationRouter(zodiosCtx, service);
+
+  const app = zodiosCtx.app();
+
+  // Disable the "X-Powered-By: Express" HTTP header for security reasons.
+  // See https://cheatsheetseries.owasp.org/cheatsheets/HTTP_Headers_Cheat_Sheet.html#recommendation_16
+  app.disable("x-powered-by");
+
+  app.use((_, res, next) => {
+    if (isFeatureFlagEnabled(config, "featureFlagNotificationConfig")) {
+      next();
+    } else {
+      res.status(403).send("Feature flag not enabled");
```

Comment:
Here we should somehow throw our `featureFlagNotEnabled` and use `makeApiProblem` to send a Problem error, otherwise this is not an error in the "standard" format we follow. See for example this similar middleware in BFF: https://github.com/pagopa/interop-be-monorepo/blob/e1fe0e3e1239e4589102bf17eb6a8e8348137fbf/packages/backend-for-frontend/src/utilities/middlewares.ts#L33